go/vt/vttablet/tabletserver/vstreamer/engine.go

```diff
@@ -138,6 +138,7 @@ func NewEngine(env tabletenv.Env, ts srvtopo.Server, se *schema.Engine, lagThrot
 		errorCounts:               env.Exporter().NewCountersWithSingleLabel("VStreamerErrors", "Tracks errors in vstreamer", "type", "Catchup", "Copy", "Send", "TablePlan"),
 		vstreamerFlushedBinlogs:   env.Exporter().NewCounter("VStreamerFlushedBinlogs", "Number of times we've successfully executed a FLUSH BINARY LOGS statement when starting a vstream"),
 	}
+	env.Exporter().NewGaugeFunc("VStreamerCount", "Current number of vstreamers", vse.getVStreamerCount)
```

Comment:
Can we instead use a `stats.Gauge` as a new field `vstreamerCount` in `vstreamer.Engine`? Then in `vse.Stream()`: `vse.vstreamerCount.Add(1)` while creating a stream and `vse.vstreamerCount.Add(-1)`. `stats.Gauge` is an `atomicInt64`. We then avoid the mutex requirement.

Reply:
@rohit-nayak-ps good idea, I'll make that change 👍

Reply:
@rohit-nayak-ps / @mattlord changes made in https://github.com/vitessio/vitess/pull/11978/commits/5d889e0f2ca1a71303052171daa712e7b6ba271f